pkg/prometheus/resource_selector.go

```diff
@@ -225,55 +219,75 @@ func validateScrapeIntervalAndTimeout(p monitoringv1.PrometheusInterface, scrape
 	return CompareScrapeTimeoutToScrapeInterval(scrapeTimeout, scrapeInterval)
 }
 
+func validateRelabelConfigs(p monitoringv1.PrometheusInterface, rcs []*monitoringv1.RelabelConfig) error {
+	for i, rc := range rcs {
+		if rc == nil {
+			return fmt.Errorf("null relabel config")
+		}
+
+		if err := validateRelabelConfig(p, *rc); err != nil {
+			return fmt.Errorf("[%d]: %w", i, err)
+		}
+	}
+
+	return nil
+}
+
 func validateRelabelConfig(p monitoringv1.PrometheusInterface, rc monitoringv1.RelabelConfig) error {
 	relabelTarget := regexp.MustCompile(`^(?:(?:[a-zA-Z_]|\$(?:\{\w+\}|\w+))+\w*)+$`)
 	promVersion := operator.StringValOrDefault(p.GetCommonPrometheusFields().Version, operator.DefaultPrometheusVersion)
+
 	version, err := semver.ParseTolerant(promVersion)
 	if err != nil {
 		return errors.Wrap(err, "failed to parse Prometheus version")
 	}
+
 	minimumVersionCaseActions := version.GTE(semver.MustParse("2.36.0"))
 	minimumVersionEqualActions := version.GTE(semver.MustParse("2.41.0"))
+	if rc.Action == "" {
+		rc.Action = string(relabel.Replace)
```

Comment:
shouldn't this be already taken care since we set default as `replace`?
https://github.com/prometheus-operator/prometheus-operator/blob/78a28753bef7f702c27f16ea38449f599d805301/pkg/apis/monitoring/v1/prometheus_types.go#L1340-L1341

Reply:
This is for tests + environments that don't have a recent version of the CRD (e.g. one without the default value).